packages/flutter/lib/src/painting/basic_types.dart

```diff
@@ -269,3 +269,27 @@ bool axisDirectionIsReversed(AxisDirection axisDirection) {
   }
   return null;
 }
+
+/// Different ways to clip for a widget
+enum ClipOption {
+  /// No clip at all
+  none,
+
+  /// Fastest clip without anti-aliasing. Reasonable fidelity.
+  /// This will be the common case when you clip rectangles or shapes with very
+  /// small curved areas (such as rounded rectangles with small corners).
+  hardEdge,
+
+  /// A little slower Clip. Better fidelity. It has smoother and anti-aliased
+  /// edges at a slightly higher cost. This will be the common case when dealing
+  /// with circles and arcs.
+  antiAlias,
+
+  /// Very slow. Highest fidelity and slightly different semantics.
+  /// This will be only rarely needed. One case where you might need this is if
+  /// you have an image overlaid on a very different background color. In these
+  /// cases, consider whether you can avoid overlaying multiple colors in one
+  /// spot (e.g. by having the background color only present where the image is
+  /// absent).
+  antiAliasWithSaveLayer
```

Comment:
for these, we should follow the following convention (used elsewhere in the docs):

```
/// Brief sentence fragment that says what it does.
///
/// This is a longer sentence that describes the semantics of the value in detail,
/// including edge cases. There might be more sentences here.
///
/// Then, have a sentence or more than talks about the current performance
/// implications.
///
/// Finally, a brief comment if the feature is unlikely to be used.
///
/// See also:
///
///  * [SomethingElse], which is something relevant.
```

The "see also" block would be omitted if not useful. But e.g. `antiAlias` might point to `Canvas.antiAlias`.

So for example:

```
/// Clip, but do not apply anti-aliasing.
///
/// This mode enables clipping, but curves and non-axis-aligned straight lines will be
/// jagged as no effort is made to anti-alias.
///
/// Faster than other clipping modes, but slower than [none].
///
/// This is a reasonable choice when clipping is needed, if the container is an axis-
/// aligned rectangle or an axis-aligned rounded rectangle with very small corner radii.
///
/// See also:
///
///  * [antiAlias], which is more reasonable when clipping is needed and the shape is not
///    an axis-aligned rectangle.
hardEdge,
```

Reply:
Thank you @Hixie for the sample writing! However, I couldn't find `Canvas.antiAlias`. It seems that we don't offer non-anti-aliased draw in `Canvas`?

Reply:
Ah it's on Paint: https://master-docs-flutter-io.firebaseapp.com/flutter/dart-ui/Paint/isAntiAlias.html